lib/src/Uncertainty/Distribution/JointDistribution.cxx

```diff
@@ -344,17 +387,17 @@ JointDistribution * JointDistribution::clone() const
 Point JointDistribution::getRealization() const
 {
   const UnsignedInteger dimension = getDimension();
+  // Complex case: use 1D CDF inversion
+  if (!kernel_.isCopula() || !hasIndependentCopula())
+  {
+    Point realization(kernel_.getRealization());
+    for (UnsignedInteger i = 0; i < dimension; ++i) realization[i] = distributionCollection_[i].computeScalarQuantile(realization[i]);
+    return realization;
+  } // General case
   if (dimension == 1) return distributionCollection_[0].getRealization();
```

Comment:
Shouldn't this case be handled at the start of getRealization, since it is the trivial case?